dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityQueryParams.java

```diff
@@ -638,10 +630,7 @@ public TrackedEntityQueryParams orderBy(String field, SortDirection direction) {
    */
   public TrackedEntityQueryParams orderBy(TrackedEntityAttribute tea, SortDirection direction) {
     this.order.add(new Order(tea, direction));
-    this.addFiltersIfNotExist(
-        QueryItem.getQueryItems(List.of(tea)).stream()
-            .filter(sAtt -> !this.getFilters().contains(sAtt))
-            .toList());
+    this.filters.putIfAbsent(tea, new ArrayList<>());
```

Comment:
> To make it work properly and differentiate the first 2 cases, we need to represent the presence of the filter with no values as a null. So there will be an entry in the filters map with the attribute UID and a list of filters with one null element.

These many cases we have to deal with make it complex 😅 We want an API that is safe to use. I wonder if we now depend on order of calling `filterBy`, `orderBy`.

Do we get the same behavior when calling
* `orderBy(uid1, DESC)` and then `filterBy(uid1)` or
* `filterBy(uid1)` and then `orderBy(uid1, DESC)`

if not I think we should make it so it does not matter so its safer to use.

Reply:
Absolutely, I think this can be problematic, I will check.
I remember that this was one of my concerns but then we went to all-devs-week and I completely forgot to check :grimacing: